ingestion/src/metadata/ingestion/source/bigquery.py

```diff
@@ -145,6 +148,25 @@ def standardize_schema_table_names(
             raise ValueError(f"schema {schema} does not match table {table}")
         return segments[0], segments[1]
 
+    def fetch_sample_data(self, schema: str, table: str) -> Optional[TableData]:
+        resp_sample_data = super().fetch_sample_data(schema, table)
+        if not resp_sample_data and self.config.partition_query:
+            try:
+                logger.info("Using Query for Partitioned Tables")
```

Comment:
@ayush-shah Thanks for the quick fix on this. However we should use the partition key to only to look at the latest data. Example if the partition is daily lets look at the a day before partition to get the sample data. Otherwise this query can be running long and expensive

Reply:
second thing can be part of another PR is to fetch the partition key details and we will add backend support for this to add the partition key details as well